### PR TITLE
Override keyword

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: c++/src/H5DataSpace.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ class H5_DLLCPP DataSpace : public IdComponent {
     DataSpace &operator=(const DataSpace &rhs);
 
     // Closes this dataspace.
-    virtual void close();
+    virtual void close() override;
```
Comment (qkoziol) [inline c++/src/H5DataSpace.h]: Since you've defined the H5_OVERRIDE macro, do you want to use it in the code?

Comment (seanm) [reply c++/src/H5DataSpace.h]: Yes. But as I said in https://github.com/HDFGroup/hdf5/issues/334 `H5Cpp.h` doesn't work for me, it seems to not be included by any of the headers where I added override. In fact these headers don't have any `#include` statements at all, so I'm a bit confused where to put `H5_OVERRIDE`.

Comment (seanm) [reply c++/src/H5DataSpace.h]: OK, I defined `H5_OVERRIDE` in `H5Include.h`.  Seems like a correct enough place.  There's a similar block there for ancient C++ that has no `bool` keyword (!).

**Comment by seanm** [comment]
@bmribler does defining H5_OVERRIDE in H5Include.h seem ok to you?

**Review by byrnHDF — APPROVED** [review]

**Review by bmribler — APPROVED** [review]

**Comment by seanm** [comment]
@lrknox could this one go into 1.10 branch please?  It causes annoying warnings in VTK & ITK CI.

**Comment by lrknox** [comment]
@seanm I think so.  I created pull requests today to merge previous PRs to develop #s 425, 427, 428, 434, and 437 to hdf5_1_12 and PRs #427, 428, 434 and 437 to hdf5_1_10.  #425 will go to 1.10 separately due to conflicts, and we’ll discuss 1.8 which has more conflicts.

I suppose I can go ahead and create the PRs to merge #429, 433, and 449 to hdf5_1_12 and hdf5_1_10 pending develop branch testing tomorrow.

**Comment by seanm** [comment]
@lrknox I don't need every PR I've created to be in 1.10, just the ones I mention.  But of course backport them if you want. :)

**Comment by seanm** [comment]
@lrknox was this one backported to 1.12 branch?

**Comment by seanm** [comment]
@lrknox what is the name of the branch that represents the tip of 1.12.x?  i.e. what will become the next 1.12 release? Is it`1.12/master` or `hdf5_1_12`?

**Comment by lrknox** [comment]
hdf5_1_12 is the development branch for future 1.12.x releases.

1.12/master holds code from previous releases,

We will create a release preparation branch hdf5_1_12_2 from the hdf5_1_12 branch shortly before the next (1.12.2) release.

Pull requests should go to the develop branch first, then to the hdf5_1_12 branch.




________________________________

**Pull request merged**